modules/pulsar/src/main/java/org/testcontainers/containers/PulsarContainer.java

```diff
@@ -41,36 +51,65 @@ public PulsarContainer(String pulsarVersion) {
     public PulsarContainer(final DockerImageName dockerImageName) {
         super(dockerImageName);
         dockerImageName.assertCompatibleWith(DockerImageName.parse("apachepulsar/pulsar"));
-
         withExposedPorts(BROKER_PORT, BROKER_HTTP_PORT);
-        withCommand("/pulsar/bin/pulsar", "standalone", "--no-functions-worker", "-nss");
-        waitingFor(Wait.forHttp(METRICS_ENDPOINT).forStatusCode(200).forPort(BROKER_HTTP_PORT));
     }
 
     @Override
     protected void configure() {
         super.configure();
-
-        if (functionsWorkerEnabled) {
-            withCommand("/pulsar/bin/pulsar", "standalone");
-            waitingFor(
-                new WaitAllStrategy()
-                    .withStrategy(waitStrategy)
-                    .withStrategy(Wait.forLogMessage(".*Function worker service started.*", 1))
-            );
-        }
+        setupCommandAndEnv();
     }
 
     public PulsarContainer withFunctionsWorker() {
         functionsWorkerEnabled = true;
         return this;
     }
 
+    public PulsarContainer withTransactions() {
+        transactions = true;
+        return this;
+    }
+
+    public PulsarContainer withConfiguration(String name, String value) {
+        return withEnv("PULSAR_PREFIX_" + name, value);
+    }
+
+    public PulsarContainer withConfiguration(Map<String, String> configuration) {
+        configuration.forEach((name, value) -> withEnv("PULSAR_PREFIX_" + name, value));
+        return this;
+    }
+
     public String getPulsarBrokerUrl() {
         return String.format("pulsar://%s:%s", getHost(), getMappedPort(BROKER_PORT));
     }
 
     public String getHttpServiceUrl() {
         return String.format("http://%s:%s", getHost(), getMappedPort(BROKER_HTTP_PORT));
     }
+
+    protected void setupCommandAndEnv() {
+        String standaloneBaseCommand =
+            "/pulsar/bin/apply-config-from-env.py /pulsar/conf/standalone.conf " + "&& bin/pulsar standalone";
+
+        if (!functionsWorkerEnabled) {
+            standaloneBaseCommand += " --no-functions-worker -nss";
+        }
+
+        withCommand("/bin/bash", "-c", standaloneBaseCommand);
+        if (transactions) {
+            withConfiguration("transactionCoordinatorEnabled", "true");
+        }
+
+        if (functionsWorkerEnabled) {
```

Comment:
Is this hierarchy for defining the wait strategy correct? So if `functionsWorkerEnabled` and `transactionsEnabled`, it is enough to wait for `Wait.forLogMessage(".*Function worker service started.*", 1)`? Or would we need a `WaitAllStrategy` that combines the waiting aspects?

Reply:
also, setting the defaultStrategy, sounds redundant to me.

Reply:
`WaitAllStrategy` seems the best option, thanks.

@eddumelendez can you explain better? what are you suggesting?

Reply:
default strategy is looking is just waiting for the default port configured and later other strategies have been added which already cover the listening port plus the readiness of a specific path. I would like to understand why the defaultStrategy was added